### PR TITLE
deps: Upgrade to Node v16, in setup docs and in CI

### PR DESCRIPTION
**Pull request by gnprice**
I've been using this for local development for a few weeks now,
and all seems well.  (That's longer than it needs to be; I just
hadn't come back to this to follow up.)

Node v12 is going EOL in just over a week, 2022-04-30, so it's
past time to make this upgrade official.

Also delete an old troubleshooting entry for a problem with Node v11
that was avoided by sticking to v10; and fix two other references to
Node v10 (where the problem was using an older version like v8) so
that they remain accurate now and for further upgrades in the future.

(At least until Node v1000.  On Node's current release cadence, that
will happen in a little under 500 years.  I think we can tolerate
having to tweak this check then.)

Fixes: #4263

### CODE REVIEW

**Comment by gnprice** [comment]
Should wait to merge this one until CI passes, because this is a rare PR that touches the CI workflow config directly.

File: tools/test
Comment (andersk) [inline tools/test:195]: `v[1-9][0-9].* | v[1-9][0-9][0-9]*` handles the v1000 problem.

But there’s a more standard solution: [`engines`](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines) in `package.json`.

Comment (gnprice) [reply tools/test:195]: Great, yeah, in reality it works. If I set it to `>=20`:
```
$ time yarn install
yarn install v1.22.17
[1/5] Validating package.json...
error ZulipMobile@1.0.0: The engine "node" is incompatible with this module. Expected version ">=20". Got "14.18.3"
error Found incompatible module.
```

(I guess these docs are only claiming to document `npm` anyway, not `yarn`. Trying `npm install`, it seems to hang -- or maybe just takes a long time without producing much output -- and I don't feel like digging into it.)

**Comment by gnprice** [comment]
CI passed; and did indeed [get Node v16](https://github.com/zulip/zulip-mobile/runs/6136074009?check_suite_focus=true):
```
Run actions/setup-node@v2
Attempt to resolve the latest version from manifest...
Resolved as '16.14.2'
Found in cache @ /opt/hostedtoolcache/node/16.14.2/x64
```

**Comment by gnprice** [comment]
(BTW I did post a revision applying Anders's feedback. I see I hadn't posted a comment saying so.)

File: docs/howto/build-run.md
Comment (andersk) [inline docs/howto/build-run.md:55]: Does this mean the latest Yarn 1.x (classic) version, by the way?

**Review by andersk — APPROVED** [review]
Not sure if you were waiting for my review, but LGTM.

**Comment by chrisbobbe** [comment]
It also LGTM. 🙂

**Comment by gnprice** [comment]
Thanks for the reviews! Merged.

File: docs/howto/build-run.md
Comment (gnprice) [reply docs/howto/build-run.md:55]: It does. I guess it'd be good to say that explicitly, though it looks like that URL continues to be about the classic versions.

Also I guess at some point we should look at migrating to the "modern" versions. For a while there it looked like a complex migration for something that wouldn't ultimately work for our form of deployment, and it seemed like the developers didn't even really understand why everything hadn't moved to their new model. But it looks like they now promise:
https://yarnpkg.com/getting-started/migration#step-by-step
that the new model isn't required, and that Yarn v2+ should be an improvement even while continuing to use `node_modules`.

Comment (gnprice) [reply docs/howto/build-run.md:55]: Noted now: https://github.com/zulip/zulip-mobile/commit/75f20bf2e00ecf181906f693b90b9884c283b9c5

**Pull request merged**
